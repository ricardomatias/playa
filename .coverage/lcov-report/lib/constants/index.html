<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/constants</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/constants
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>56/64</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>56/63</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="chords.js"><a href="chords.js.html">chords.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="intervals.js"><a href="intervals.js.html">intervals.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="note.js"><a href="note.js.html">note.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="rhythmsDistributions.js"><a href="rhythmsDistributions.js.html">rhythmsDistributions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="scales.js"><a href="scales.js.html">scales.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="ticks.js"><a href="ticks.js.html">ticks.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="20" class="abs medium">12/20</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="63.16" class="pct medium">63.16%</td>
	<td data-value="19" class="abs medium">12/19</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 14:43:58 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
