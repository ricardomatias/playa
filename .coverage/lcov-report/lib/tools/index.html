<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/tools</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/tools
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>296/325</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>102/120</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>50/59</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.76% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>275/303</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="choose.js"><a href="choose.js.html">choose.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="distance.js"><a href="distance.js.html">distance.js</a></td>
	<td data-value="97.37" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.37" class="pct high">97.37%</td>
	<td data-value="38" class="abs high">37/38</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="97.14" class="pct high">97.14%</td>
	<td data-value="35" class="abs high">34/35</td>
	</tr>

<tr>
	<td class="file high" data-value="distribute.js"><a href="distribute.js.html">distribute.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	</tr>

<tr>
	<td class="file high" data-value="euclidean.js"><a href="euclidean.js.html">euclidean.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="friendly.js"><a href="friendly.js.html">friendly.js</a></td>
	<td data-value="97.56" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.56" class="pct high">97.56%</td>
	<td data-value="82" class="abs high">80/82</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="97.4" class="pct high">97.4%</td>
	<td data-value="77" class="abs high">75/77</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="interval.js"><a href="interval.js.html">interval.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="36" class="abs high">32/36</td>
	<td data-value="85" class="pct high">85%</td>
	<td data-value="20" class="abs high">17/20</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="91.43" class="pct high">91.43%</td>
	<td data-value="35" class="abs high">32/35</td>
	</tr>

<tr>
	<td class="file high" data-value="random.js"><a href="random.js.html">random.js</a></td>
	<td data-value="93.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="ring.js"><a href="ring.js.html">ring.js</a></td>
	<td data-value="77.27" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 77%;"></div><div class="cover-empty" style="width:23%;"></div></div></td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	</tr>

<tr>
	<td class="file high" data-value="roll.js"><a href="roll.js.html">roll.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="rotate.js"><a href="rotate.js.html">rotate.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="shuffle.js"><a href="shuffle.js.html">shuffle.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="time.js"><a href="time.js.html">time.js</a></td>
	<td data-value="71.93" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.93" class="pct medium">71.93%</td>
	<td data-value="57" class="abs medium">41/57</td>
	<td data-value="67.86" class="pct medium">67.86%</td>
	<td data-value="28" class="abs medium">19/28</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="56" class="abs medium">40/56</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
