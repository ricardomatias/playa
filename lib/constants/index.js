/**
 * Constants
 * @namespace {module} Constants
 */
export * from './chords';
export * from './scales';
export * from './note';
export * from './turns';
export * from './time';
export { default as MIDI_NOTES } from './midi';
export { default as TICKS } from './ticks';
export { default as INTERVALS } from './intervals';
