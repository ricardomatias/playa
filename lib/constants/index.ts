/**
 * Constants
 * @namespace {module} Constants
 */
export * from './chords';
export * from './scales';
export * from './note';
export * from './turns';
export * from './midi';
export * from './ticks';
export * from './intervals';
