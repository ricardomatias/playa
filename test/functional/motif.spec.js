import * as R from 'ramda';
import Scale from '../../lib/core/Scale';
import Chord from '../../lib/core/Chord';
import createMotif from '../../lib/functional/motif';
import { distribute } from '@ricardomatias/roll';
import { calcDuration } from '../../lib/utils';
import { Rhythm, choose } from '../../lib/tools';
import Random from '../../lib/tools/random';
import { TICKS } from '../../lib/constants';

const { mixed } = Rhythm.presets;

const ONE_BAR = TICKS.get('1n');


describe('A Motif test suite', () => {
	it('should generate motif based on scale', () => {
		// given
		Random.setSeed('test');

		const scale = new Scale('A', Scale.MAJOR);

		let rhythm = Rhythm.free('1:0:0', mixed, [], distribute.decreasing);

		// when
		let motif = createMotif(scale.notes, rhythm);

		// then
		expect(motif).toMatchSnapshot();
		expect(calcDuration(motif)).toEqual(ONE_BAR);

		// when
		Random.setSeed('test2');

		rhythm = Rhythm.free('2:0:0', mixed, [], distribute.decreasing);

		motif = createMotif(scale.notes, rhythm);

		// then
		expect(motif).toMatchSnapshot();
		expect(calcDuration(motif)).toEqual(ONE_BAR * 2);
	});


	it('should generate createMotif based on chord', () => {
		// given
		Random.setSeed('test');

		const chord = new Chord('Dbm6');
		let rhythm = Rhythm.free('1:0:0', mixed, [], distribute.decreasing);

		// when
		let motif = createMotif(chord.notes, rhythm);

		expect(motif).toMatchSnapshot();
		expect(calcDuration(motif)).toEqual(ONE_BAR);

		// when
		Random.setSeed('test2');

		rhythm = Rhythm.free('2:0:0', mixed, [], distribute.decreasing);

		motif = createMotif(chord.notes, rhythm);

		expect(motif).toMatchSnapshot();
		expect(calcDuration(motif)).toEqual(ONE_BAR * 2);
	});


	it('should return pattern with time = start time', () => {
		// given
		Random.setSeed('test');

		const chord = new Chord('Dbm6');
		let rhythm = Rhythm.free('0:1:0', mixed);

		// when
		let motif = createMotif(chord.notes, rhythm, 0);

		expect(motif).toMatchSnapshot();

		rhythm = Rhythm.free('0:3:0', mixed);

		motif = createMotif(chord.notes, rhythm, 480);

		expect(motif).toMatchSnapshot();
	});

	it('should generate motif with rests', () => {
		// given
		Random.setSeed('test');

		const scale = new Scale('A', Scale.MAJOR);
		const durations = choose(mixed, Random.int(R.length(mixed)));

		const rhythm = Rhythm.free('1:0:0', mixed, durations, distribute.decreasing);

		// when
		const motif = createMotif(scale.notes, rhythm);

		// then
		expect(motif).toMatchSnapshot();
		expect(calcDuration(motif)).toEqual(ONE_BAR);
	});
});
