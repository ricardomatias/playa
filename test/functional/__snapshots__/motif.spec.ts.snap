// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Motif test suite should generate createMotif based on chord 1`] = `
Array [
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 64,
    "next": 480,
    "note": "E3",
    "time": 0,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "midi": 68,
    "next": 640,
    "note": "Ab3",
    "time": 480,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 68,
    "next": 880,
    "note": "Ab3",
    "time": 640,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 68,
    "next": 1000,
    "note": "Ab3",
    "time": 880,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 64,
    "next": 1320,
    "note": "E3",
    "time": 1000,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 64,
    "next": 1560,
    "note": "E3",
    "time": 1320,
  },
  Object {
    "dur": 360,
    "isRest": false,
    "midi": 61,
    "next": 1920,
    "note": "Db3",
    "time": 1560,
  },
]
`;

exports[`A Motif test suite should generate createMotif based on chord 2`] = `
Array [
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 64,
    "next": 480,
    "note": "E3",
    "time": 0,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 64,
    "next": 720,
    "note": "E3",
    "time": 480,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 64,
    "next": 960,
    "note": "E3",
    "time": 720,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 68,
    "next": 1440,
    "note": "Ab3",
    "time": 960,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "midi": 64,
    "next": 1600,
    "note": "E3",
    "time": 1440,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 64,
    "next": 2080,
    "note": "E3",
    "time": 1600,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 61,
    "next": 2400,
    "note": "Db3",
    "time": 2080,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 68,
    "next": 2640,
    "note": "Ab3",
    "time": 2400,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 68,
    "next": 3120,
    "note": "Ab3",
    "time": 2640,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 64,
    "next": 3360,
    "note": "E3",
    "time": 3120,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 68,
    "next": 3600,
    "note": "Ab3",
    "time": 3360,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 68,
    "next": 3720,
    "note": "Ab3",
    "time": 3600,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 68,
    "next": 3840,
    "note": "Ab3",
    "time": 3720,
  },
]
`;

exports[`A Motif test suite should generate motif based on scale 1`] = `
Array [
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 73,
    "next": 480,
    "note": "C#4",
    "time": 0,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "midi": 76,
    "next": 640,
    "note": "E4",
    "time": 480,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 78,
    "next": 880,
    "note": "F#4",
    "time": 640,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 76,
    "next": 1000,
    "note": "E4",
    "time": 880,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 74,
    "next": 1320,
    "note": "D4",
    "time": 1000,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 73,
    "next": 1560,
    "note": "C#4",
    "time": 1320,
  },
  Object {
    "dur": 360,
    "isRest": false,
    "midi": 69,
    "next": 1920,
    "note": "A3",
    "time": 1560,
  },
]
`;

exports[`A Motif test suite should generate motif based on scale 2`] = `
Array [
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 73,
    "next": 480,
    "note": "C#4",
    "time": 0,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 74,
    "next": 720,
    "note": "D4",
    "time": 480,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 71,
    "next": 960,
    "note": "B3",
    "time": 720,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 76,
    "next": 1440,
    "note": "E4",
    "time": 960,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "midi": 73,
    "next": 1600,
    "note": "C#4",
    "time": 1440,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 71,
    "next": 2080,
    "note": "B3",
    "time": 1600,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 71,
    "next": 2400,
    "note": "B3",
    "time": 2080,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 76,
    "next": 2640,
    "note": "E4",
    "time": 2400,
  },
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 76,
    "next": 3120,
    "note": "E4",
    "time": 2640,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 74,
    "next": 3360,
    "note": "D4",
    "time": 3120,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 76,
    "next": 3600,
    "note": "E4",
    "time": 3360,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 76,
    "next": 3720,
    "note": "E4",
    "time": 3600,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 78,
    "next": 3840,
    "note": "F#4",
    "time": 3720,
  },
]
`;

exports[`A Motif test suite should generate motif with given durations 1`] = `
Array [
  Object {
    "dur": 480,
    "isRest": false,
    "midi": 71,
    "next": 480,
    "note": "B3",
    "time": 0,
  },
  Object {
    "dur": 960,
    "isRest": false,
    "midi": 80,
    "next": 1440,
    "note": "G#4",
    "time": 480,
  },
  Object {
    "dur": 360,
    "isRest": false,
    "midi": 74,
    "next": 1800,
    "note": "D4",
    "time": 1440,
  },
]
`;

exports[`A Motif test suite should generate motif with rests 1`] = `
Array [
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 76,
    "next": 120,
    "note": "E4",
    "time": 0,
  },
  Object {
    "dur": 200,
    "isRest": true,
    "midi": -1,
    "next": 320,
    "note": "",
    "time": 120,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 73,
    "next": 440,
    "note": "C#4",
    "time": 320,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 74,
    "next": 560,
    "note": "D4",
    "time": 440,
  },
  Object {
    "dur": 120,
    "isRest": true,
    "midi": -1,
    "next": 680,
    "note": "",
    "time": 560,
  },
  Object {
    "dur": 400,
    "isRest": true,
    "midi": -1,
    "next": 1080,
    "note": "",
    "time": 680,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 74,
    "next": 1400,
    "note": "D4",
    "time": 1080,
  },
  Object {
    "dur": 240,
    "isRest": true,
    "midi": -1,
    "next": 1640,
    "note": "",
    "time": 1400,
  },
  Object {
    "dur": 120,
    "isRest": false,
    "midi": 74,
    "next": 1760,
    "note": "D4",
    "time": 1640,
  },
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 76,
    "next": 2080,
    "note": "E4",
    "time": 1760,
  },
]
`;

exports[`A Motif test suite should return pattern with time = start time 1`] = `
Array [
  Object {
    "dur": 320,
    "isRest": false,
    "midi": 68,
    "next": 800,
    "note": "Ab3",
    "time": 480,
  },
  Object {
    "dur": 720,
    "isRest": false,
    "midi": 68,
    "next": 1520,
    "note": "Ab3",
    "time": 800,
  },
  Object {
    "dur": 240,
    "isRest": false,
    "midi": 64,
    "next": 1760,
    "note": "E3",
    "time": 1520,
  },
  Object {
    "dur": 160,
    "isRest": false,
    "midi": 68,
    "next": 1920,
    "note": "Ab3",
    "time": 1760,
  },
]
`;
